book_extractors/common/extractors/tests/test_extraction_pipeline.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+from book_extractors.common.extractors.base_extractor import BaseExtractor
+from book_extractors.extraction_pipeline import ExtractionPipeline, configure_extractor
+
+'''
+TODO: Populate this file with many more tests for extraction_pipeline
+'''
+
+
+class TestExtractionPipelineResultsPassing:
+    def should_contain_parent_results_deep_in_the_pipeline(self):
+        test_pipeline = ExtractionPipeline([
+            configure_extractor(MockExtractor),
+            configure_extractor(MockExtractorAWithSubpipeline)
+        ])
+        parent_results = {
+            'extraction_results': {
+                'result': 'test'
+            },
+            'parent_results': {
+                'extraction_results': {
+                    'name': {
+                        'gender': 'Male'
+                    },
+                    'test_result': 'testiness',
+                    'farmDetails': {
+                        'farmTotalArea': 57.0
+                    }
+                },
+                'parent_results': None
+            }
+        }
+
+        test_pipeline.process({'text': 'test string'}, parent_pipeline_results=parent_results)
```

Comment:
Saisiko tämän tehtyä jotenkin niin, että nuo mock extractorit palauttavat jotakin järkevää tänne ylimmän pipelinen lopputulokseen, ja sieltä saadun arvon perusteella voitaisiin tässä assertata jotakin järkevää?

Toinen vaihtoehto, joka on kenties parempi on virittää tähän sellainen spy, joka vakoilee relevantteja funkkareita tuolta extractorista. Tietokanta reposta löytyy testi nimeltä `should_not_do_anything_for_children_if_they_have_not_changed` jossa on käytetty spy:tä tsekkaamaan montako kertaa tiettyä funktiota on kutsuttu.

Ehkä tän voisi jotenkin vakoojillakin setviä, mutta menee kyllä hankalaksi. Ohan tää joo vähän hassu näin, mut en nyt suoriltaan keksi miten tän sais tuolla lopussa assertattua. Täähän on oikeestaan integraatiotesti, eikä yksikkötesti, joten periaatteessa oikee tapa ois virittää nää extractorit siten, että lopuksi pipelinesta tulee oikee data ulos. Tai vähän kuin yksikkötesti syvälle pipelinelle, jossa katot että onko se ihan lopputulos oikea. Sen voinee tosin tehdä vasta lopuksi, kun tää koko systeemi toimii jotenkin järkevästi.

Eli no voi antaa olla näin (uudelleen kirjoita toi kommentti vähän lyhyemmäksi ja revieweer maininta pois :)), voit urheilla tässä vaiheessa tai vain kirjoittaa lopuksi tuollaisen kuvaamani kaltaisen ylimmän tason testin.